amqp-java-demo/src/main/java/ProducerTest.java

```diff
@@ -0,0 +1,91 @@
+import com.rabbitmq.client.BasicProperties;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.ConfirmCallback;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import com.rabbitmq.client.MessageProperties;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Collections;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentNavigableMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.concurrent.TimeoutException;
+
+public class ProducerTest {
+    private static final String QUEUE_NAME = "yunQi-queue";
+    private static final String EXCHANGE_NAME = "yunQi-exchange";
+    private static final String ROUTING_KEY = "yunQi-routing-key";
+    private static final String VHOST = "yunQi-vhost";
+    private static final String CONNECTION_NAME = "yunQi-connection";
+    private static final String HOSTNAME = "your-instance-endpoint"; // 替换为配置的阿里云MQ实例的公网接入点
+    private static final String USERNAME = "your-username"; // 替换为配置的静态用户名
+    private static final String PASSWORD = "your-password"; // 替换为配置的静态密码
+
+    public static void main(String[] argv) throws Exception {
+        ConnectionFactory factory = new ConnectionFactory();
+        factory.setHost(HOSTNAME);
+        factory.setUsername(USERNAME);
+        factory.setPassword(PASSWORD);
+        factory.setVirtualHost(VHOST);
+        factory.setClientProperties(Collections.singletonMap("connection_name", CONNECTION_NAME));
+        try (Connection connection = factory.newConnection();
+             Channel channel = connection.createChannel()) {
+            ConcurrentNavigableMap<Long/*deliveryTag*/, String/*msgId*/> outstandingConfirms=new ConcurrentSkipListMap<>();
+            initChannel(channel,outstandingConfirms);
+            channel.queueDeclare(QUEUE_NAME, true, false, false, null);
+            channel.exchangeDeclare(EXCHANGE_NAME, "direct", true); // 根据实际情况更改交换类型
+            // 开始发送消息，3600条消息，每条发送后暂停1秒，将持续1小时。
+            int messageCount = 3600;
+            for (int i = 1; i <= messageCount; i++) {
+                try {
+                    String message = "消息Body-" + i;
+                    String msgId=UUID.randomUUID().toString();
+                    AMQP.BasicProperties properties = new AMQP.BasicProperties.Builder()
+                        .messageId(msgId)
+                        .build();
+                    outstandingConfirms.put(channel.getNextPublishSeqNo(), msgId);
+                    channel.basicPublish(EXCHANGE_NAME, ROUTING_KEY, properties, message.getBytes(StandardCharsets.UTF_8));
+                    Thread.sleep(1000L);
+                } catch (Exception e) {
+                    System.out.println(" Send fail, error: " + e.getMessage());
+                    Thread.sleep(5000L);
```

Comment:
If the channel or connection is unavailable, sleeping for 5 seconds will still result in failure. You should set the automaticRecovery parameter to true.